custom-recipes/googlesheets-append/recipe.json

```diff
@@ -27,12 +27,34 @@
     ],
 
     "params": [
+        {
+            "name": "auth_type",
+            "label": "Type of authentication",
+            "type": "SELECT",
+            "selectChoices": [
+                {
+                    "value": "service-account",
+                    "label": "Service account preset"
+                },
+                {
+                    "value": "credentials",
+                    "label": "Service account credentials (legacy)"
+                }
+            ]
+        },
         {
             "name": "credentials",
             "label": "Service Account credentials",
             "description": "Copy/paste the JSON private key provided by Google, or a path to a file containing the key. Click on the link above above to read how to get it.",
             "type": "TEXTAREA",
-            "mandatory" : true
+            "visibilityCondition": "model.auth_type == 'credentials' || model.auth_type.mode == 'NONE'"
```

Comment:
I think the name change I suggest above would make this visibility condition a bit more meaningful as it would read

``` 
"model.auth_type == 'legacy-service-account' || model.auth_type.mode == 'NONE'"
```

Reply:
Done in [0ae3fbb](https://github.com/dataiku/dss-plugin-googlesheets/pull/2/commits/0ae3fbb57d6aee92b5fdfdcf37504229875a8a26)